frontend/js/src/playlists/utils.tsx

```diff
@@ -130,3 +131,19 @@ export function listenToJSPFTrack(listen: Listen): JSPFTrack {
       : undefined,
   };
 }
+
+export function listenOrJSPFTrackToQueueItem(
+  track: Listen | JSPFTrack
+): BrainzPlayerQueueItem {
+  let listenTrack: Listen;
+  if (has(track, "title")) {
+    listenTrack = JSPFTrackToListen(track as JSPFTrack);
+  } else {
+    listenTrack = cloneDeep(track as BrainzPlayerQueueItem);
+  }
+  const queueItem = {
+    ...listenTrack,
+    id: `${listenTrack.track_metadata.track_name} - ${Date.now().toString()}`,
```

Comment:
If this util is mainly to add an `id` , I wonder if instead we can directly pass an `id` prop to the QueueItemCard. Do you reckon that would work?